src/main/java/com/leetcode/practice/easy/binary_search/BinarySearch.java

```diff
@@ -0,0 +1,27 @@
+package com.leetcode.practice.easy.binary_search;
+
+public class BinarySearch {
+    public int findElementInArray(int[] nums, int target) {
+        int start=0;
+        int end=nums.length-1;
+        int mid;
+        if(nums.length==0)
```

Comment:
corner cases check should come first even before declaring any variables. Everything else go after that.

These ifs can be structured properly. All the -1 cases as or condition and 0 case as separate if